### ISSUE TITLE
Add PSR-16 compliance

### ISSUE THREAD

**Issue by jstewmc** [jstewmc/gravity#5]
We should implment the [PSR-16](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-16-simple-cache.md) interface in order to increase the interoperability of our cache.

**Comment by javabudd** [comment]
We should reopen this ticket, looking at the interface documentation gives advice as to what should be done when the argument type is not valid. 

Documentation for get():

```
    /**
     * Fetches a value from the cache.
     *
     * @param string $key     The unique key of this item in the cache.
     * @param mixed  $default Default value to return if the key does not exist.
     *
     * @return mixed The value of the item from the cache, or $default in case of cache miss.
     *
     * @throws \Psr\SimpleCache\InvalidArgumentException
     *   MUST be thrown if the $key string is not a legal value.
     */
```

**Comment by jstewmc** [comment]
Yeah, I like dat. Re-opened bb!

**Comment by jstewmc** [comment]
Closed by #38.
